--- projects/libcxx/lib/CMakeLists.txt_orig	2014-05-19 14:42:05.000000000 +0200
+++ projects/libcxx/lib/CMakeLists.txt	2014-05-19 15:50:25.000000000 +0200
@@ -50,13 +50,20 @@
 append_if(link_flags LIBCXX_HAS_NODEFAULTLIBS_FLAG -nodefaultlibs)
 
 if ( APPLE )
+  Set(LIBCXX_INSTALL_PATH "/usr/lib" CACHE STRING "Some user-specified option")
+  Set(LIBCXX_LIBCXXABI_LIBRARY_PATH "/usr/lib" CACHE STRING "Some user-specified option")
+
+  If(NOT ${LIBCXX_LIBCXXABI_LIBRARY_PATH} STREQUAL "/usr/lib")
+    Set(CMAKE_OSX_SYSROOT "")
+  EndIf()
+
   if ( CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "10.6" )
     list(APPEND compile_flags "-U__STRICT_ANSI__")
     list(APPEND link_flags
       "-compatibility_version 1"
       "-current_version ${LIBCXX_VERSION}"
-      "-install_name /usr/lib/libc++.1.dylib"
-      "-Wl,-reexport_library,/usr/lib/libc++abi.dylib"
+      "-install_name ${LIBCXX_INSTALL_PATH}/libc++.1.dylib"
+      "-Wl,-reexport_library,${LIBCXX_LIBCXXABI_LIBRARY_PATH}/libc++abi.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "/usr/lib/libSystem.B.dylib")
   else()
@@ -64,19 +71,19 @@
       list(FIND ${CMAKE_OSX_ARCHITECTURES} "armv7" OSX_HAS_ARMV7)
       if (OSX_HAS_ARMV7)
         set(OSX_RE_EXPORT_LINE
-          "${CMAKE_OSX_SYSROOT}/usr/lib/libc++abi.dylib"
+          "${CMAKE_OSX_SYSROOT}/${LIBCXX_LIBCXXABI_LIBRARY_PATH}/libc++abi.dylib"
           "-Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++sjlj-abi.exp")
       else()
         set(OSX_RE_EXPORT_LINE
-          "-Wl,-reexport_library,${CMAKE_OSX_SYSROOT}/usr/lib/libc++abi.dylib")
+          "-Wl,-reexport_library,${CMAKE_OSX_SYSROOT}/${LIBCXX_LIBCXXABI_LIBRARY_PATH}/libc++abi.dylib")
       endif()
     else()
-      set (OSX_RE_EXPORT_LINE "/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++abi${LIBCXX_LIBCPPABI_VERSION}.exp")
+      set (OSX_RE_EXPORT_LINE "-Wl,-reexport_library,${LIBCXX_LIBCXXABI_LIBRARY_PATH}/libc++abi.dylib -Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++abi${LIBCXX_LIBCPPABI_VERSION}.exp")
     endif()
 
     list(APPEND link_flags
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
+      "-install_name ${LIBCXX_INSTALL_PATH}/libc++.1.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "${OSX_RE_EXPORT_LINE}"
       "-Wl,-force_symbols_not_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/notweak.exp"
