--- tools/clang/tools/extra/clang-tidy/ClangTidy.cpp
+++ tools/clang/tools/extra/clang-tidy/ClangTidy.cpp
@@ -58,7 +58,7 @@ static const StringRef StaticAnalyzerChecks[] = {
 #define GET_CHECKERS
 #define CHECKER(FULLNAME, CLASS, DESCFILE, HELPTEXT, GROUPINDEX, HIDDEN)       \
   FULLNAME,
-#include "../../../lib/StaticAnalyzer/Checkers/Checkers.inc"
+#include "../../../tools/clang/lib/StaticAnalyzer/Checkers/Checkers.inc"
 #undef CHECKER
 #undef GET_CHECKERS
 };
