--- projects/libcxxabi/src/CMakeLists.txt.orig	2015-09-16 21:09:05.000000000 +0200
+++ projects/libcxxabi/src/CMakeLists.txt	2015-09-16 21:11:28.000000000 +0200
@@ -67,13 +67,14 @@
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
       "-current_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib")
     list(APPEND LIBCXXABI_LINK_FLAGS
         "/usr/lib/libSystem.B.dylib")
   else()
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib"
+      "-nodefaultlibs")
   endif()
 endif()
 
@@ -101,6 +102,7 @@
       OUTPUT_NAME   "c++abi"
       VERSION       "1.0"
       SOVERSION     "1"
+      INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
     )
   list(APPEND LIBCXXABI_TARGETS "cxxabi_shared")
 endif()
