diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index afc388e..f79f948 100644
--- projects/libcxx/lib/CMakeLists.txt
+++ projects/libcxx/lib/CMakeLists.txt
@@ -110,8 +110,8 @@ if ( APPLE AND (LIBCXX_CXX_ABI_LIBNAME STREQUAL "libcxxabi" OR
     add_link_flags(
       "-compatibility_version 1"
       "-current_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
-      "-Wl,-reexport_library,/usr/lib/libc++abi.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++.1.dylib"
+      "-Wl,-reexport_library,${CMAKE_INSTALL_PREFIX}/lib/libc++abi.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "/usr/lib/libSystem.B.dylib")
   else()
@@ -126,16 +126,18 @@ if ( APPLE AND (LIBCXX_CXX_ABI_LIBNAME STREQUAL "libcxxabi" OR
           "-Wl,-reexport_library,${CMAKE_OSX_SYSROOT}/usr/lib/libc++abi.dylib")
       endif()
     else()
-      set(OSX_RE_EXPORT_LINE "/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++abi${LIBCXX_LIBCPPABI_VERSION}.exp")
+      get_property(abiloc TARGET cxxabi_shared PROPERTY LOCATION)
+      set(OSX_RE_EXPORT_LINE "-Wl,-reexport_library,${abiloc}")
     endif()
 
     add_link_flags(
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++.1.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "${OSX_RE_EXPORT_LINE}"
       "-Wl,-force_symbols_not_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/notweak.exp"
-      "-Wl,-force_symbols_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/weak.exp")
+      "-Wl,-force_symbols_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/weak.exp"
+      "-nodefaultlibs")
   endif()
 endif()
 
@@ -150,6 +152,7 @@ set_target_properties(cxx
     OUTPUT_NAME   "c++"
     VERSION       "${LIBCXX_ABI_VERSION}.0"
     SOVERSION     "${LIBCXX_ABI_VERSION}"
+    INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
   )
 
 if (LIBCXX_ENABLE_EXPERIMENTAL_LIBRARY)
