--- projects/libcxx/lib/CMakeLists.txt.orig	2015-09-16 21:35:42.000000000 +0200
+++ projects/libcxx/lib/CMakeLists.txt	2015-09-16 21:38:06.000000000 +0200
@@ -84,8 +84,8 @@
     list(APPEND LIBCXX_LINK_FLAGS
       "-compatibility_version 1"
       "-current_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
-      "-Wl,-reexport_library,/usr/lib/libc++abi.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++.1.dylib"
+      "-Wl,-reexport_library,${CMAKE_INSTALL_PREFIX}/lib/libc++abi.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "/usr/lib/libSystem.B.dylib")
   else()
@@ -96,16 +96,15 @@
           "${CMAKE_OSX_SYSROOT}/usr/lib/libc++abi.dylib"
           "-Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++sjlj-abi.exp")
       else()
-        set(OSX_RE_EXPORT_LINE
-          "-Wl,-reexport_library,${CMAKE_OSX_SYSROOT}/usr/lib/libc++abi.dylib")
+        set(OSX_RE_EXPORT_LINE "-Wl,-reexport_library,${CMAKE_INSTALL_PREFIX}/lib/libc++abi.dylib")
       endif()
     else()
-      set (OSX_RE_EXPORT_LINE "/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++abi${LIBCXX_LIBCPPABI_VERSION}.exp")
+      set(OSX_RE_EXPORT_LINE "-Wl,-reexport_library,${CMAKE_INSTALL_PREFIX}/lib/libc++abi.dylib")
     endif()
 
     list(APPEND LIBCXX_LINK_FLAGS
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++.1.dylib"
       "-Wl,-unexported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp"
       "${OSX_RE_EXPORT_LINE}"
       "-Wl,-force_symbols_not_weak_list,${CMAKE_CURRENT_SOURCE_DIR}/notweak.exp"
@@ -123,6 +122,7 @@
     OUTPUT_NAME   "c++"
     VERSION       "1.0"
     SOVERSION     "1"
+    INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
   )
 
 install(TARGETS cxx
