--- projects/compiler-rt/CMakeLists.txt_orig 2014-05-13 10:23:48.000000000 +0200
+++ projects/compiler-rt/CMakeLists.txt 2014-05-13 10:24:11.000000000 +0200
@@ -188,9 +188,9 @@
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
   set(SANITIZER_COMMON_SUPPORTED_DARWIN_OS osx)
-  if (IOSSIM_SDK_DIR)
-    list(APPEND SANITIZER_COMMON_SUPPORTED_DARWIN_OS iossim)
-  endif()
+#  if (IOSSIM_SDK_DIR)
+#    list(APPEND SANITIZER_COMMON_SUPPORTED_DARWIN_OS iossim)
+#  endif()
 
   if(COMPILER_RT_USES_LIBCXX)
     set(SANITIZER_MIN_OSX_VERSION 10.7)
--- tools/clang/lib/Driver/Tools.cpp_orig	2014-05-22 13:59:55.131399151 +0200
+++ tools/clang/lib/Driver/Tools.cpp	2014-05-22 14:04:14.291931576 +0200
@@ -191,6 +191,14 @@
 
   // LIBRARY_PATH - included following the user specified library paths.
   addDirectoryList(Args, CmdArgs, "-L", "LIBRARY_PATH");
+
+  // Add -L<prefix>/lib to default linker arguments. This is needed to
+  // find libc++ automatically.
+  // On Linux add also -rpath.
+  CmdArgs.push_back(Args.MakeArgString("-L" + D.InstalledDir + "/../lib")); 
+  if (TC.getTriple().getOS() == llvm::Triple::Linux) {
+    CmdArgs.push_back(Args.MakeArgString("-rpath=" + D.InstalledDir + "/../lib"));
+  }
 }
 
 /// \brief Determine whether Objective-C automated reference counting is
